src/encoded/commands/release_file.py

```diff
@@ -0,0 +1,491 @@
+            self.add_release_item_to_patchdict(
+                analyte, f"Analyte - {analyte[PC.SUBMITTED_ID]}"
+            )
+
+            for sample_uuid in analyte[PC.SAMPLES]:
+                sample = self.get_metadata(sample_uuid)
+                self.add_release_item_to_patchdict(
+                    sample, f"Sample - {sample[PC.SUBMITTED_ID]}"
+                )
+
+                sample_sources = sample[PC.SAMPLE_SOURCES]
+                for sample_source_id in sample_sources:
+                    sample_source = self.get_metadata(sample_source_id)
+                    self.add_release_item_to_patchdict(
+                        sample_source,
+                        f"SampleSource - {sample_source[PC.SUBMITTED_ID]}",
+                    )
+
+        print("\nThe following metadata patches will be carried out in the next step:")
+        for info in self.patch_infos:
+            print(info)
+
+        if len(self.warnings) > 0:
+            print(
+                f"\n{bcolors.WARNING}Please note the following warnings:{bcolors.ENDC}"
+            )
+            for warning in self.warnings:
+                print(warning)
+
+    def execute(self):
+        print("Validating all patch dictionaries...")
+        try:
+            for patch_dict in self.patch_dicts:
+                ff_utils.patch_metadata(
+                    patch_dict,
+                    obj_id=patch_dict[PC.UUID],
+                    add_on="?check_only=true",
+                    key=self.key,
+                )
+        except Exception as e:
+            print(str(e))
+            self.print_error_and_exit("Validation failed.")
+
+        print("Validation done. Patching...")
+        try:
+            for patch_dict in self.patch_dicts:
+                ff_utils.patch_metadata(
+                    patch_dict,
+                    obj_id=patch_dict[PC.UUID],
+                    key=self.key,
+                )
+        except Exception as e:
+            print(str(e))
+            self.print_error_and_exit("Patching failed.")
+
+        print(
+            f"{bcolors.OKGREEN}Release of File {self.file_accession} completed.{bcolors.ENDC}"
+        )
+
+    def show_patch_dicts(self):
+        print("\n")
+        pp.pprint(self.patch_dicts)
+
+    def add_release_item_to_patchdict(self, item: Dict, item_desc: str):
+        """Sets the status of the item to released and
+        adds the corresponding patch dict
+
+        Args:
+            item (Dict): Portal item
+            item_desc (str): Just used for generating more usefuls patch infos
+        """
+        self.patch_infos.append(f"\n{item_desc} ({item[PC.ACCESSION]}):")
+
+        if item[PC.STATUS] == PC.RELEASED:
+            self.patch_infos.append(
+                f"  - {bcolors.OKBLUE}{PC.STATUS}{bcolors.ENDC} is already set to {bcolors.OKBLUE}{PC.RELEASED}{bcolors.ENDC}. Not patching."
+            )
+            return
+
+        patch_body = {
+            PC.UUID: item[PC.UUID],
```

Comment:
Seems like the UUID is being added here only for the item identifier to be used later. Could consider using a separate data structure, like a named tuple or something, for greater clarity, but should be fine as is.